### PR TITLE
FIX: update voter information upon remote change

### PR DESCRIPTION
**Pull request by merefield**
Fixes issue with polls not being fully updated by remote vote contributions in (semi-) real-time.

This was down to too great a focus on tracking local state and not accommodating a more data down approach with responsive getters. 

This is now implemented.

I've tried hard to minimise the changes whilst making sure the paradigm is properly followed through.

### CODE REVIEW

File: plugins/poll/lib/ranked_choice.rb [hunk 1]
```diff
@@ -115,7 +115,7 @@ def self.find_potential_winners(tally, max_votes)
   def self.majority_check(tally, potential_winners, max_votes)
     total_votes = tally.values.sum
 
-    max_votes > total_votes / 2 || potential_winners.count == 1
+    max_votes && (max_votes > total_votes / 2 || potential_winners.count == 1)
```
[the remark this answers is not in the document]
Comment (merefield) [reply plugins/poll/lib/ranked_choice.rb:118]: https://github.com/discourse/discourse/blob/87450f677b32093d57d3b44592e5caecbf2f1f8e/plugins/poll/lib/ranked_choice.rb#L41

you cannot vote unless there is at least one valid selection, so there should always be a non-nil `max_votes`

not a huge deal though :)

[the remark this answers is not in the document]
Comment (jjaffeux) [reply plugins/poll/lib/ranked_choice.rb]: https://github.com/user-attachments/assets/2bf638b5-8be5-4bed-a5c2-48336a6b3db4


We talked of this in DM, this will be fixed in separate PR

**Review by jjaffeux — APPROVED** [review]

**Review by merefield — COMMENTED** [review]
@jjaffeux, I don't your change will work.  `preloadedVoters` needs to be tracked because it is locally managed.  The data is coming from both the direct upstream via widget interface and also the `fetchVoters` API call.

**Review by merefield — COMMENTED** [review]
@jjaffeux why is this necessary - this was working fine?

File: plugins/poll/assets/javascripts/discourse/components/poll.gjs [hunk 2]
```diff
@@ -253,7 +253,7 @@ export default class PollComponent extends Component {
   }
 
   get voters() {
-    return this.poll.voters;
+    return this.poll.get("voters");
```
Comment (merefield) [inline plugins/poll/assets/javascripts/discourse/components/poll.gjs:254]: For my own benefit, why is this change necessary - it was working fine?

Comment (jjaffeux) [reply plugins/poll/assets/javascripts/discourse/components/poll.gjs:254]: it was not working fine in my tests :)

**Comment by jjaffeux** [comment]
> @jjaffeux, I don't your change will work. `preloadedVoters` needs to be `tracked` because it is locally managed and written to. The data is coming from both the direct upstream via widget interface and also the `fetchVoters` API call (and that's why a test is now failing). This is why it is first defaulted then tracked. It is updated on the fly using the modifier deliberately.

Yes I tried to refactor more of prelaodedVoters locally for this case, but it's just too much to change
